engine/new_runtime/src/memories/noncontiguous_memory.rs

```diff
@@ -0,0 +1,372 @@
+use crate::memories::buffer::Buffer;
+use crate::memories::file_memory::FileMemory;
+use crate::crypto_utils::crypto_box::{BoxProvider};
+use crate::locked_memory::ProtectedConfiguration::*;
+use crate::locked_memory::ProtectedMemory;
+use crate::locked_memory::LockedConfiguration::{self, *};
+use crate::locked_memory::LockedMemory;
+use crate::locked_memory::MemoryError::{self, *};
+use crate::types::Bytes;
+use crypto::hashes::sha;
+use core::fmt::{self, Debug, Formatter};
+use zeroize::Zeroize;
+
+
+static IMPOSSIBLE_CASE: &'static str = "NonContiguousMemory: this case should not happen if allocated properly";
+
+// Currently we only support data of 32 bytes in noncontiguous memory
+const NC_DATA_SIZE: usize = 32;
```

Comment:
Does this restriction have a reason, or is it just easier to partition / reassemble 32 byte chunks? I wonder, if this could be more flexible, but it's not a requirement. A possible extension could be a String of non-contiguous memory.

Reply:
The only hard requirement is that NC memory should have the same size of the output of the hashing algorithm used. 
In our case NC memory will be used for keys which are 32 bytes in the current implementation. 
I agree that it should and could be more flexible.